### ISSUE TITLE
rust-analyzer failed to run build scripts.

### ISSUE THREAD

**Issue by marhkb** [bilelmoussaoui/flatpak-vscode#169]
Hi,

I remember that in earlier versions, the rust-analyzer binary used to be set in `settings_json.json` like
```json
{
    "files.watcherExclude": {
        "**/.git/objects/**": true,
        "**/.git/subtree-cache/**": true,
        "**/node_modules/*/**": true,
        "**/.hg/store/**": true,
        "**/.flatpak": true,
        ".flatpak": true,
        ".flatpak/**": true,
        "_build/**": true
    },
    "rust-analyzer.server.path": "/var/home/marcus/Entwicklung/pods/.flatpak/rust-analyzer.sh",
    "rust-analyzer.files.excludeDirs": [
        ".flatpak"
    ],
    "rust-analyzer.runnables.extraArgs": [
        "--target-dir=_build/src"
    ],
    "rust-analyzer.runnables.command": "/var/home/marcus/Entwicklung/pods/.flatpak/cargo.sh"
}
```

This doesn't seem the case any more when I set up a new project. I noticed that rust-analyzer now fails to run the build scripts. It seems it isn't executed in the flatpak environment?

```
[ERROR rust_analyzer::lsp_utils] rust-analyzer failed to run build scripts

The following warnings were emitted during compilation:


error: failed to run custom build command for `pango-sys v0.15.10`

Caused by:
  process didn't exit successfully: `/var/home/marcus/Entwicklung/telegrand1/target/debug/build/pango-sys-7a7e5a927336befc/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=PANGO_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=`"pkg-config" "--libs" "--cflags" "pango" "pango >= 1.49.2"` did not exit successfully: exit status: 1
  error: could not find system library 'pango' required by the 'pango-sys' crate

  --- stderr
  Package dependency requirement 'pango >= 1.49.2' could not be satisfied.
  Package 'pango' has version '1.48.11', required version is '>= 1.49.2'

The following warnings were emitted during compilation:


The following warnings were emitted during compilation:


The following warnings were emitted during compilation:


error: build failed
```

Old projects that have the settings and scripts inside the `.flatpak` folder still work as expected.
Am I doing something wrong, or is this a regression?

Visual Studio Code Info
```
Version: 1.67.0
Commit: 57fd6d0195bb9b9d1b49f6da5db789060795de47
Datum: 2022-05-04T12:08:42.978Z
Electron: 17.4.1
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
Betriebssystem: Linux x64 5.17.9-300.fc36.x86_64
```

Related extensions
```
flatpak-vscode: v0.0.27
rust-analyzer: v0.3.1075
```

**Comment by marhkb** [comment]
I'm not sure whether the upgrade from 0.0.26 to 0.0.27 caused this. I've just noticed it now.

> Can you remove the config file

I removed the whole `.vscode` folder.

> run the clean command  

Output of `Flatpak: Clean up the build directories`
```
>>> Deleted Flatpak repository directory
>>> Deleted _build directory
>>> Pipeline state reset
```

>  reload the window

`Developer: Reload window` re-creates the `.vscode` folder and the `settings.json` inside it. Its content is
```json
{
    "files.watcherExclude": {
        "**/.git/objects/**": true,
        "**/.git/subtree-cache/**": true,
        "**/node_modules/*/**": true,
        "**/.hg/store/**": true,
        ".flatpak/**": true,
        "_build/**": true
    }
}
```

But rust-analyzer throws the error as soon as I open a `*.rs` file. I guess the `rust-analyzer.*` settings are supposed to be there?

**Comment by bilelmoussaoui** [comment]
If they are not, then your manifest is not properly detected somehow. Could you paste the manifest here?

**Comment by marhkb** [comment]
```json
{
    "app-id": "com.github.melix99.telegrand.Devel",
    "runtime": "org.gnome.Platform",
    "runtime-version": "42",
    "sdk": "org.gnome.Sdk",
    "sdk-extensions": [
        "org.freedesktop.Sdk.Extension.rust-stable",
        "org.freedesktop.Sdk.Extension.llvm13"
    ],
    "command": "telegrand",
    "finish-args": [
        "--socket=fallback-x11",
        "--socket=wayland",
        "--share=network",
        "--share=ipc",
        "--device=dri",
        "--talk-name=org.a11y.Bus",
        "--env=G_MESSAGES_DEBUG=none",
        "--env=RUST_BACKTRACE=1"
    ],
    "build-options": {
        "append-path": "/usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/llvm13/bin",
        "env": {
            "CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER": "clang",
            "CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUSTFLAGS": "-C link-arg=-fuse-ld=/usr/lib/sdk/rust-stable/bin/mold"
        },
        "build-args": [
            "--share=network"
        ]
    },
    "cleanup": [
        "/include",
        "/lib/cmake",
        "/lib/pkgconfig",
        "*.a"
    ],
    "modules": [
        {
            "name": "gtk",
            "buildsystem": "meson",
            "sources": [
                {
                    "type": "git",
                    "url": "https://gitlab.gnome.org/GNOME/gtk.git",
                    "tag": "4.6.4"
                },
                {
                    "type" : "patch",
                    "path" : "gtk-reversed-list.patch"
                }
            ]
        },
        {
            "name": "tdlib-prebuilt",
            "only-arches": [ "x86_64", "aarch64" ],
            "buildsystem": "simple",
            "build-commands": [
                "install -d /app/lib/pkgconfig",
                "install -D libtdjson.so.* /app/lib",
                "ln -sf /app/lib/libtdjson.so.* /app/lib/libtdjson.so",
                "install -D pkgconfig/* /app/lib/pkgconfig"
            ],
            "sources": [
                {
                    "only-arches": [ "x86_64" ],
                    "type": "archive",
                    "url": "https://github.com/melix99/tdjson-ci/releases/download/1.8.2-1/tdjson-x86_64.zip",
                    "sha256": "2efddcf0a7d94321b08b449ab9ff7e5b1faabe7ee9f666ddc4d8a4f21cadbec4",
                    "strip-components": 0
                },
                {
                    "only-arches": [ "aarch64" ],
                    "type": "archive",
                    "url": "https://github.com/melix99/tdjson-ci/releases/download/1.8.2-1/tdjson-aarch64.zip",
                    "sha256": "401ceae0a9b838d5b6e273ba9d84d64cf8b20842fbefc51d4856b3f34cc010d7",
                    "strip-components": 0
                }
            ]
        },
        {
            "name": "tdlib",
            "skip-arches": [ "x86_64", "aarch64" ],
            "buildsystem": "cmake-ninja",
            "builddir": true,
            "config-opts": [
                "-DCMAKE_BUILD_TYPE=Release"
            ],
            "sources": [
                {
                    "type": "git",
                    "url": "https://github.com/tdlib/td.git",
                    "commit": "3f54c301ead1bbe6529df4ecfb63c7f645dd181c"
                }
            ]
        },
        {
            "name": "telegrand",
            "buildsystem": "meson",
            "run-tests": true,
            "config-opts": [
                "-Dprofile=development"
            ],
            "sources": [
                {
                    "type": "dir",
                    "path": "../"
                }
            ]
        }
    ]
}

```

**Comment by bilelmoussaoui** [comment]
That should work fine, is it the upstream manifest of telegrand? I will give it a try in a bit.

**Comment by SeaDve** [comment]
> I guess the rust-analyzer.* settings are supposed to be there?

Yep it's suppose to be there unless the integration failed to load.

**Comment by marhkb** [comment]
> That should work fine, is it the upstream manifest of telegrand? I will give it a try in a bit.

Yes, it is the upstream manifest. I also tried to start over by deleting all vscode user data, but still without success.

> Yep it's suppose to be there unless the integration failed to load somewhere (which should show error)

I have the issue on 2 computers. Maybe it's a Fedora Silverblue problem?
I tried to use the vscode rpm as an overlay, but still have no luck.

**Comment by bilelmoussaoui** [comment]
Ah I see, the rust-analyzer extension changed it id.

**Comment by bilelmoussaoui** [comment]
Fixed in 0.0.28, should be up in a couple of minutes
